starknet_py/tests/e2e/client/websocket_client_test.py

```diff
@@ -107,3 +128,230 @@ async def test_unsubscribe_with_non_existing_id(
+    )
+
+    unsubscribe_result = await websocket_client.unsubscribe(subscription_id)
+    assert unsubscribe_result is True
+
+
+@pytest.mark.asyncio
+async def test_subscribe_new_transaction_receipts_with_finality_status(
+    websocket_client: WebsocketClient,
+    deployed_balance_contract,
+    argent_account_v040: BaseAccount,
+):
+    receipts: List[TransactionReceipt] = []
+
+    def handler(new_tx_receipt: NewTransactionReceiptsNotification):
+        nonlocal receipts
+        receipts.append(new_tx_receipt.result)
+
+    subscription_id = await websocket_client.subscribe_new_transaction_receipts(
+        handler=handler,
+        sender_address=[argent_account_v040.address],
+        finality_status=[TransactionFinalityStatusWithoutL1.ACCEPTED_ON_L2],
+    )
+
+    increase_balance_call = Call(
+        to_addr=deployed_balance_contract.address,
+        selector=get_selector_from_name("increase_balance"),
+        calldata=[100],
+    )
+    execute = await argent_account_v040.execute_v3(
+        calls=increase_balance_call, resource_bounds=MAX_RESOURCE_BOUNDS
+    )
+
+    await argent_account_v040.client.wait_for_tx(tx_hash=execute.transaction_hash)
+    await asyncio.sleep(10)
+
+    assert len(receipts) >= 1
+    assert all(
+        r.finality_status == TransactionFinalityStatus.ACCEPTED_ON_L2 for r in receipts
+    )
+
+    unsubscribe_result = await websocket_client.unsubscribe(subscription_id)
+    assert unsubscribe_result is True
+
+
+@pytest.mark.asyncio
+async def test_subscribe_events_with_all_filters(
+    client: FullNodeClient,
+    websocket_client: WebsocketClient,
+    deployed_balance_contract,
+    argent_account_v040: BaseAccount,
+):
+    received_events: List = []
+
+    def handler(new_events_notification: NewEventsNotification):
+        nonlocal received_events
+        received_events.append(new_events_notification.result)
+
+    latest_block = await client.get_block(block_hash="latest")
+    assert isinstance(latest_block, StarknetBlock)
+
+    subscription_id = await websocket_client.subscribe_events(
+        handler=handler,
+        from_address=argent_account_v040.address,
+        keys=[[]],
+        block_number=latest_block.block_number,
+        finality_status=TransactionFinalityStatusWithoutL1.ACCEPTED_ON_L2,
+    )
+
+    increase_balance_call = Call(
+        to_addr=deployed_balance_contract.address,
+        selector=get_selector_from_name("increase_balance"),
+        calldata=[100],
+    )
+    execute = await argent_account_v040.execute_v3(
+        calls=increase_balance_call, resource_bounds=MAX_RESOURCE_BOUNDS
+    )
+
+    await argent_account_v040.client.wait_for_tx(tx_hash=execute.transaction_hash)
+    await asyncio.sleep(3)
+
+    assert len(received_events) >= 1
+    assert all(
+        ev.finality_status == TransactionFinalityStatus.ACCEPTED_ON_L2
+        for ev in received_events
+    )
```

Comment:
Maybe we should also check if `from_address` and `block_number` filters were fullfiled?